exporter/exporterhelper/queued_retry_experimental.go

```diff
@@ -44,8 +44,9 @@ type QueueSettings struct {
 	NumConsumers int `mapstructure:"num_consumers"`
 	// QueueSize is the maximum number of batches allowed in queue at a given time.
 	QueueSize int `mapstructure:"queue_size"`
-	// PersistentStorageEnabled describes whether persistence via a file storage extension is enabled
-	PersistentStorageEnabled bool `mapstructure:"persistent_storage_enabled"`
+	// StorageID if not empty, enables the persistent storage and uses the component specified
+	// as a storage extension for the persistent queue
+	StorageID *config.ComponentID `mapstructure:"storage"`
```

Comment:
@djaglowski I saw a PR from you related to this, can you ensure we are consistent here? Another idea is to extract this into a "configstorage" similar with "configauth" to ensure everyone uses this in the same way.

Reply:
Yes, for reference [here is the related PR on contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12871/).

It looks like we have a difference in behavior regarding whether or not storage is automatically used if there is exactly one storage extension. This was discussed [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10915#issuecomment-1154055913). Basically, there is a desire to have this functionality for easier configuration in a common case and for backwards compatibility.

Reply:
> Basically, there is a desire to have this functionality for easier configuration in a common case and for backwards compatibility.

I don't think a behavior like that is desired for new components, I don't disagree that for backwards compatibility you may do that, but I disagree to do that for new components. The main reason is that someone comes, adds a new extension that implements the storage extension and everything breaks now.

Reply:
Although I do not want to introduce a breaking change, I think in this case we should consider it in order to keep behavior the same for all components that use storage extensions. To my knowledge, this would only affect the stanza-based and otlpjsonfile receivers, none of which are beyond alpha status.